code/parachain/frame/composable-traits/src/lib.rs

```diff
@@ -34,6 +33,12 @@
 	trivial_numeric_casts,
 	unused_extern_crates
 )]
+#![allow(incomplete_features)]
+#![feature(associated_type_defaults)]
+#![feature(trait_alias)]
+#![feature(const_trait_impl)]
+#![feature(const_convert)]
+#![feature(adt_const_params)]
```

Comment:
enabling incomplete features in production code probably isn't the best idea

Reply:
in theory can make these test only and add more unwraps/if let into prod codebase

Reply:
also it will work if to encode Rational as u128 high and low bits. so it will be ugly big impl. and will not be able to use as #[const] parameter.

Reply:
Not sure what you mean by not being able to use it in a const parameter - this works: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4b669b68d44e5d039f4b034317534934

Reply:
const type parameter and #[const] config parameter

Reply:
'const type parameter' is referring to the `RationalLike` trait?

Reply:
#[pallet::const]
type FooBar = Get<Rational64>

Reply:
Not sure I understand - `Get<Rational64>` will work with what I sent before, unstable features aren't required for that

Reply:
You could also use https://docs.rs/static_assertions/latest/static_assertions/macro.const_assert_ne.html

Reply:
> Not sure I understand - Get<Rational64> will work with what I sent before, unstable features aren't required for that
My test (may be 3 month ago) shown that without const cannot have nice config. So may be did something wrong.

Reply:
other refactoring https://github.com/ComposableFi/composable/pull/2118#discussion_r1015873472